src/schema/rules/checks/nirs.yaml

```diff
@@ -0,0 +1,88 @@
+---
+NASamplingFreq:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "channels"
+    - extension == ".tsv"
+    - assoications.nirs.json.SamplingFrequency == 'n/a'
+  checks:
+    - columns.sampling_frequency != null
+
+ACCELChannelCountReq:
+  selectors:
+    - datatype == "nirs"
+    - suffix == "channels"
+    - extension == ".tsv"
+    - columns.type[0] == "ACC"
+  checks:
+    - associations.nirs.ACCELChannelCount != null
```

Comment:
Should it be >0 as well?

Reply:
you mean so that it is not a negative number?

Reply:
I think that's covered in the object definition. In this case, I was thinking that, if the "type" column has an "ACC" value in it, then there has to be at least one ACC channel. I mean, I guess it should actually check that the number of "ACC" rows in the "type" column should match the "ACCELChannelCount" value, but I don't know if that's feasible.

Reply:
I see, I am fine with both !=null and >0 ... given that negative values are forbidden, both amount to the same.

>  mean, I guess it should actually check that the number of "ACC" rows in the "type" column should match the "ACCELChannelCount" value, but I don't know if that's feasible.

that'd be a really nice future enhancement

Reply:
this is discussed below in https://github.com/bids-standard/bids-specification/pull/802#discussion_r968384253 ... resolving here.